src/handlers/location-handler.js

```diff
@@ -0,0 +1,19 @@
+const path = require('path');
+const url = require('url');
+const { fileRead }  = require('../file-access');
+
+const locationHandler = (request, response) => {
+    const filePath = path.join(__dirname,'..' ,'locations.json');
+    fileRead(filePath, request, response, (data) => {
+
+        const queryObject = url.parse(request.url, true).query;
```

Comment:
you can get the endpoint again from the reqeust object, no need to pass it as an argument. `const url = req.url` or `const { url } = req`. 

Note:  `url.parse` is legacy. To get the querystring you can do the following: 

```js
  const searchParams = new URLSearchParams(req.url.split('?')[1])
  console.log(searchParams.get('q')) // note this could be null so you might want to add a condition or a default value
```